backend/security/src/main/kotlin/io/tolgee/security/ratelimit/RateLimited.kt

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (C) 2023 Tolgee s.r.o. and contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.tolgee.security.ratelimit
+
+/**
+ * Enables rate limiting on a specific route in addition to the global rate limit.
+ * Has no effect when rate limiting is disabled in server configuration.
+ */
+@Target(AnnotationTarget.FUNCTION, AnnotationTarget.ANNOTATION_CLASS)
+annotation class RateLimited(
+  /**
+   * Maximum amount of requests that can be made during the window interval.
+   */
+  val limit: Int,
+  /**
+   * Size of the window to consider in milliseconds. Defaults to a second.
+   */
+  val windowSize: Long = 1_000,
+  /**
+   * Allows for grouping different endpoints under the same bucket. Will default to "{method} {path}".
+   */
+  val bucketName: String = "",
+  /**
+   * Number of route parameters to take into account during bucketing.
+   *
+   * For instance, GET `/v2/projects/1/activity` and `/v2/projects/2/activity` will be considered different buckets
+   * despite the route being the same (`/v2/projects/{id}/activity`).
+   */
+  val majorParametersToDiscriminate: Int = 1,
```

Comment:
I would much rather add string array of parameters, containing path variables to be used for bucketing. 

Also it's not clear what you mean by route parameters.

Reply:
Why not; my logic was to make it agnostic to route parameter names as I don't see a use-case where the first parameter wouldn't be discriminated but the second would.

"Route parameter" is a common term I'm used to; I'm unsure if there's a better word for it to be honest.

Reply:
In spring there is a annotation `@PathVariable`. So I would call it that way to not confuse the readers. 

The string array seems like much less confusing then this kind numbering. When you use numbers I have to think if you mean the variables or each segment...

Reply:
That's a fair point. The downside is that the annotation is by default configured to discriminate the first major route parameter, which is usually the desired behavior 99% of the time (see the provided example in kdoc).

Relying on explicit route names makes it more difficult to apply this default behavior, without relying on magic values...

Good point about the path variable naming. It's not a name I'm used to but Spring uses it, you're right it's more appropriate here in a Spring codebase.

Reply:
OK